### PR TITLE
Retransmit last flight when in finished

### PR DESCRIPTION
**Pull request by Sean-Der**
When FSM is in finished and sees a handshake message re-send the last flight again.

### CODE REVIEW

**Comment by Sean-Der** [comment]
@hasheddan @at-wat Looking at the spec I believe `FINISHED` can be simpler? 

```
// [RFC6347 Section-4.2.4]
//                      +-----------+
//                +---> | PREPARING | <--------------------+
//                |     +-----------+                      |
//                |           |                            |
//                |           | Buffer next flight         |
//                |           |                            |
//                |          \|/                           |
//                |     +-----------+                      |
//                |     |  SENDING  |<------------------+  | Send
//                |     +-----------+                   |  | HelloRequest
//        Receive |           |                         |  |
//           next |           | Send flight             |  | or
//         flight |  +--------+                         |  |
//                |  |        | Set retransmit timer    |  | Receive
//                |  |       \|/                        |  | HelloRequest
//                |  |  +-----------+                   |  | Send
//                +--)--|  WAITING  |-------------------+  | ClientHello
//                |  |  +-----------+   Timer expires   |  |
//                |  |         |                        |  |
//                |  |         +------------------------+  |
//        Receive |  | Send           Read retransmit      |
//           last |  | last                                |
//         flight |  | flight                              |
//                |  |                                     |
//               \|/\|/                                    |
//            +-----------+                                |
//            | FINISHED  | -------------------------------+
//            +-----------+
//                 |  /|\
//                 |   |
//                 +---+
//              Read retransmit
//           Retransmit last flight
```

**Comment by Sean-Der** [comment]
@at-wat @hasheddan I’m gonna merge these tonight! I believe they will have zero impact

File: handshaker_test.go [hunk 1]
```diff
@@ -225,8 +225,8 @@ func TestHandshaker(t *testing.T) {
 					t.Errorf("Client is not finished")
 				}
 				// there should be no `Finished` last retransmit from client
-				if cntClientFinishedLastRetransmit != 0 {
-					t.Errorf("Number of client finished last retransmit is wrong, expected: %d times, got: %d times", 0, cntClientFinishedLastRetransmit)
+				if cntClientFinishedLastRetransmit != 4 {
+					t.Errorf("Number of client finished last retransmit is wrong, expected: %d times, got: %d times", 4, cntClientFinishedLastRetransmit)
```
Comment (at-wat) [inline handshaker_test.go:229]: Is it expected?

Comment (Sean-Der) [reply handshaker_test.go:229]: @at-wat sorry I messed up, I merged the other PR before seeing this comment.

Yes I believe so! It is getting retransmitted from the remote now. Before I think it was detecting handshake complete and not doing a rtx.

**Pull request closed without merging**